### DISCUSSION TITLE
Invalid Memory Address or Nil Pointer Dereference Error Causing Function Crash

### DISCUSSION THREAD

**Discussion by divya-inaza** [apache/pulsar#20580] [Q&A]
This issue occurs when adding data to the input topic, resulting in a function crash with an error message stating "Invalid memory address or nil pointer dereference.

Attaching a screenshot for quick reference.

![image](https://github.com/apache/pulsar/assets/103034207/a13c2746-58dd-442b-b4ea-4f882bd7235c)

**Comment by tisonkun** [comment]
Without any code snippet it's very vague to understand where this stack belongs to. Also, a text format error stack is better to debug over a screenshot.

**Comment by divya-inaza** [comment]
Thanks for your response.

The issue is now fixed... It was related to the version of github.com/apache/pulsar/pulsar-function-go.
